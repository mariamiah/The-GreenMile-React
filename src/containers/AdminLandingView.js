/* eslint-disable react/prop-types */
/* eslint-disable react/require-default-props */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import AdminLanding from '../components/AdminLanding';
import Header from '../components/Header';
import Footer from '../components/Footer';
import FetchPackagesAction from '../actions/FetchPackagesAction';

export class AdminLandingView extends Component {
  componentDidMount() {
    this.props.FetchPackagesAction();
  }

  render() {
    if (this.props.package === []) return null;
    return (
      <div>
        <Header />
        {this.props.package.map(item => <AdminLanding item={item} key={item.package_id} />)}
        <Footer />
      </div>
    );
  }
}
const mapStateToProps = state => ({
  package: state.FetchPackagesReducer.Packages,

});
AdminLandingView.propTypes = {
  FetchPackagesAction: PropTypes.func.isRequired,

};
export default connect(mapStateToProps, { FetchPackagesAction })(AdminLandingView);
