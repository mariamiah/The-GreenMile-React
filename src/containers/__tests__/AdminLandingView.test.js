/* eslint-disable no-undef */
import React from 'react';
import { shallow, mount } from 'enzyme';
import { AdminLandingView } from '../AdminLandingView';

describe('<AdminLandingView />', () => {
  it('should render without crashing', () => {
    const props = {
      FetchProductsAction: jest.fn(),
      product: {
        map: jest.fn(),
      },
    };
    const wrapper = shallow(<AdminLandingView {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('calls componentDidMount() lifecycle method', () => {
    const props = {
      product: {
        map: jest.fn(),
      },
      FetchProductsAction: jest.fn(),

    };
    const spy = jest.spyOn(AdminLandingView.prototype, 'componentDidMount');
    const wrapper = mount(<AdminLandingView {...props} />);
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });
  it('displays AdminLanding if passed props', () => {
    const props = {
      product: [{}],
      FetchProductsAction: jest.fn(),
    };
    const wrapper = mount(
      <AdminLandingView {...props} key={props.product.id} />,
    );
    expect(wrapper.find('AdminLanding').exists()).toEqual(true);
  });
});
