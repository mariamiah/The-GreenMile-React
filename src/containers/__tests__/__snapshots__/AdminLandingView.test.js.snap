// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminLandingView /> should render without crashing 1`] = `ShallowWrapper {}`;
